pkg/client/connector/userd_trafficmgr/install.go

```diff
@@ -85,12 +87,29 @@ func (ki *installer) removeManagerAndAgents(c context.Context, agentsOnly bool,
 	}
 	// wait for all agents to be removed
 	wg.Wait()
+	close(webhookAgentChannel)
 
 	if !agentsOnly && len(errs) == 0 {
 		// agent removal succeeded. Remove the manager resources
 		if err := helm.DeleteTrafficManager(c, ki.ConfigFlags, ki.GetManagerNamespace()); err != nil {
 			addError(err)
 		}
+
+		// roll all agents installed by webhook
+		webhookWaitGroup := sync.WaitGroup{}
+		webhookWaitGroup.Add(len(webhookAgentChannel))
+		for agent := range webhookAgentChannel {
+			agent := agent // pin ?
+			go func() {
```

Comment:
An alternative to pinning is to use `go func(agent kates.Object) {` here and then pass the agent as an argument on line 109.